src/basic/socket-util.c

```diff
@@ -723,8 +723,12 @@ int socknameinfo_pretty(union sockaddr_union *sa, socklen_t salen, char **_ret)
 
         assert(_ret);
 
+#ifdef HAVE_IDN
         r = getnameinfo(&sa->sa, salen, host, sizeof(host), NULL, 0,
                         NI_IDN|NI_IDN_USE_STD3_ASCII_RULES);
+#else
+        r = getnameinfo(&sa->sa, salen, host, sizeof(host), NULL, 0, 0);
```

Comment:
This is OK, but I'd prefer if you defined a IDN_FLAGS a the top of the file, and kept just one `getnameinfo` call here.